### PR TITLE
Explorer: use identicon and token layout for unlisted tokens

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
Token mints that aren't listed in the spl-token-registry should have details pages that are similar to those that are listed.

#### Summary of Changes
- add jazzicon for unlisted images
- use standard token mint layout
- slight refactor account details page
- 
Fixes https://github.com/solana-labs/solana/issues/15764

### CODE REVIEW

**Comment by oJshua** [comment]
Explorer-Solana (42)

File: explorer/src/pages/AccountDetailsPage.tsx [hunk 1]
```diff
@@ -69,49 +71,77 @@ const TOKEN_TABS_HIDDEN = [
...
+}: {
+  address: string;
+  info?: CacheEntry<Account>;
+}) {
   const { tokenRegistry } = useTokenRegistry();
   const tokenDetails = tokenRegistry.get(address);
-  if (tokenDetails) {
+  const account = info?.data;
+  const data = account?.details?.data;
+  const isToken = data && data.program === "spl-token";
```
Comment (jstarry) [inline explorer/src/pages/AccountDetailsPage.tsx:117]: This should check if it's a mint account right?

File: explorer/src/pages/AccountDetailsPage.tsx [hunk 2]
```diff
@@ -69,49 +71,77 @@ const TOKEN_TABS_HIDDEN = [
...
+          <div className="avatar avatar-lg header-avatar-top">
+            {tokenDetails?.logoURI ? (
               <img
                 src={tokenDetails.logoURI}
                 alt="token logo"
                 className="avatar-img rounded-circle border border-4 border-body"
               />
-            </div>
+            ) : (
+              <Identicon
```
Comment (jstarry) [inline explorer/src/pages/AccountDetailsPage.tsx:131]: Could we also add this for the small icon in the owned tokens card too?

File: explorer/src/components/common/Identicon.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  address?: string | PublicKey;
+  style?: React.CSSProperties;
+  className?: string;
+}) {
+  const { style, className } = props;
+  const address =
+    typeof props.address === "string"
+      ? props.address
+      : props.address?.toBase58();
+  const ref = useRef<HTMLDivElement>();
```
Comment (jstarry) [inline explorer/src/components/common/Identicon.tsx:18]: With this change you won't need to do `ref as any` below
```suggestion
  const ref = useRef<HTMLDivElement>(null);
```

File: explorer/package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "coingecko-api": "^1.0.10",
     "cross-fetch": "^3.1.2",
     "humanize-duration-ts": "^2.1.1",
+    "jazzicon": "^1.5.0",
```
[the remark this answers is not in the document]
Comment (oJshua) [reply explorer/package.json:36]: Thanks, I didn't see that in the other repo.

**Comment by oJshua** [comment]
identicon-small

**Review by jstarry — APPROVED** [review]

**Pull request merged**